### PR TITLE
Kafka broker ordered delivery

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

Part of https://github.com/knative-sandbox/eventing-kafka-broker/issues/589

## Proposed Changes 

- Add doc for new feature https://github.com/knative-sandbox/eventing-kafka-broker/issues/589

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
cc @pierDipi @matzew

**Review by pierDipi — COMMENTED** [review]
/lgtm

There are some style issues, can you fix them?

File: docs/eventing/broker/kafka-broker.md [hunk 1]
```diff
@@ -293,6 +293,35 @@ To change the logging level to `DEBUG`, you need to:
     kubectl rollout restart deployment -n knative-eventing kafka-broker-dispatcher
     ```
 
+## Events Delivery order
+
+When dispatching events, Kafka Broker supports two different ordering guarantees:
+
+* `unordered`: Unordered consumer is a non-blocking consumer that potentially deliver messages unordered, while preserving proper offset management.
```
Comment (mpetason) [inline docs/eventing/broker/kafka-broker.md:300]:
```suggestion
* `unordered`: Unordered consumer is a non-blocking consumer that potentially delivers messages unordered, while preserving proper offset management. 
```

File: docs/eventing/broker/kafka-broker.md [hunk 2]
```diff
@@ -293,6 +293,35 @@ To change the logging level to `DEBUG`, you need to:
     kubectl rollout restart deployment -n knative-eventing kafka-broker-dispatcher
     ```
 
+## Events Delivery order
+
+When dispatching events, Kafka Broker supports two different ordering guarantees:
+
+* `unordered`: Unordered consumer is a non-blocking consumer that potentially deliver messages unordered, while preserving proper offset management. 
+* `ordered`: Ordered consumer is a per-partition blocking consumer that deliver messages in order.
```
Comment (mpetason) [inline docs/eventing/broker/kafka-broker.md:301]:
```suggestion
* `ordered`: Ordered consumer is a per-partition blocking consumer that delivers messages in order.
```

**Review by mpetason — COMMENTED** [review]
added suggestions that may read a little better

**Comment by slinkydeveloper** [comment]
@mpetason ping

File: docs/eventing/broker/kafka-broker.md [hunk 3]
```diff
@@ -293,6 +293,35 @@ To change the logging level to `DEBUG`, you need to:
     kubectl rollout restart deployment -n knative-eventing kafka-broker-dispatcher
     ```
 
+## Events Delivery order
+
+When dispatching events, Kafka Broker supports two different ordering guarantees:
```
Comment (abrennan89) [inline docs/eventing/broker/kafka-broker.md:298]:
```suggestion
When dispatching events, Kafka broker supports two different ordering guarantees:
```

File: docs/eventing/broker/kafka-broker.md [hunk 4]
```diff
@@ -293,6 +293,35 @@ To change the logging level to `DEBUG`, you need to:
...
+
+* `unordered`: Unordered consumer is a non-blocking consumer that potentially delivers messages unordered, while preserving proper offset management.
+* `ordered`: Ordered consumer is a per-partition blocking consumer that delivers messages in order.
```
Comment (abrennan89) [inline docs/eventing/broker/kafka-broker.md:301]: we call these "ordering guarantees" above, but then "consumers" here. Can we choose a single term here for referring to this or explain the relationship better?

File: docs/eventing/broker/kafka-broker.md [hunk 5]
```diff
@@ -293,6 +293,35 @@ To change the logging level to `DEBUG`, you need to:
...
+  annotations:
+     kafka.eventing.knative.dev/delivery.order: ordered
+spec:
+  broker: my-kafka-broker
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: my-service
+```
```
Comment (abrennan89) [inline docs/eventing/broker/kafka-broker.md:323]: I'd move this to the start and explain what ordered and unordered mean after it's shown where they are configured to provide context

File: docs/eventing/broker/kafka-broker.md [hunk 6]
```diff
@@ -293,6 +293,35 @@ To change the logging level to `DEBUG`, you need to:
...
+## Events Delivery order
+
+When dispatching events, Kafka Broker supports two different ordering guarantees:
+
+* `unordered`: Unordered consumer is a non-blocking consumer that potentially delivers messages unordered, while preserving proper offset management.
+* `ordered`: Ordered consumer is a per-partition blocking consumer that delivers messages in order.
+
+`unordered` is the default ordering guarantee, while **`ordered` is considered unstable, use with caution**.
+
+You can configure the delivery order guarantee using the `kafka.eventing.knative.dev/delivery.order` annotation on the `Trigger` instance.
```
Comment (abrennan89) [inline docs/eventing/broker/kafka-broker.md:305]:
```suggestion
You can configure the delivery order of events using the `kafka.eventing.knative.dev/delivery.order` annotation on the `Trigger` object.
```

File: docs/eventing/broker/kafka-broker.md [hunk 7]
```diff
@@ -293,6 +293,35 @@ To change the logging level to `DEBUG`, you need to:
...
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: my-service
+```
+
 ### Additional information
 
 - To report bugs or add feature requests, open an issue in the [eventing-kafka-broker repository](https://github.com/knative-sandbox/eventing-kafka-broker).
```
Comment (abrennan89) [inline docs/eventing/broker/kafka-broker.md:327]:
```suggestion
- To report a bug or request a feature, open an issue in the [`eventing-kafka-broker` repository](https://github.com/knative-sandbox/eventing-kafka-broker).
```

File: docs/eventing/broker/kafka-broker.md (hunk 4 above)
Comment (slinkydeveloper) [reply docs/eventing/broker/kafka-broker.md:301]: I've added _consumer_ above to refer to the fact that we're talking about the kafka broker consumer delivery guarantees

File: docs/eventing/broker/kafka-broker.md (hunk 5 above)
Comment (slinkydeveloper) [reply docs/eventing/broker/kafka-broker.md]: How about now?

**Review by abrennan89 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**